fix(ios): memory leak in require

janvennemann opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26783

Adds a missing `autorelease` for the returned `ResolvedModule *` instance that is returned with a +1 reference count.

janvennemann committed:
fix(ios): memory leak in require

build commented:
<!--
  1 failure:  :microscope: Ther...
  0 warning: 
  2 messages
  
  DangerID: danger-id-default;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :microscope: There are library changes, but no changes to the unit tests. That's OK as long as you're refactoring existing code, but will require an admin to merge this PR. Please see <a href="https://github.com/janvennemann/titanium_mobile/blob/TIMOB-26783/README.md#unit-tests">README.md#unit-tests</a> for docs on unit testing.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-10659/1/artifact/dist/mobilesdk-8.1.0.v20190130101423-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 2990 tests are passing.
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3c65289f6a16bd9f3b1f5836a02dffc6a5cd9606
</p>

janvennemann committed:
send autorelease on alloc

This also shuts up the static analyzer since the methods are not adhering to the naming convention for methods that return +1 references.

vijaysingh-axway reviewed (approved):
CR passed.

sgtcoolguy committed:
Merge branch 'master' into TIMOB-26783
